runtime/parser2/expression.go

```diff
@@ -985,22 +985,35 @@ func defineMemberExpression() {
 	setExprLeftDenotation(
 		lexer.TokenDot,
 		func(p *parser, token lexer.Token, left ast.Expression) ast.Expression {
-			return parseMemberAccess(p, left, false)
+			return parseMemberAccess(p, token, left, false)
 		},
 	)
 
 	setExprLeftBindingPower(lexer.TokenQuestionMarkDot, exprLeftBindingPowerAccess)
 	setExprLeftDenotation(
 		lexer.TokenQuestionMarkDot,
 		func(p *parser, token lexer.Token, left ast.Expression) ast.Expression {
-			return parseMemberAccess(p, left, true)
+			return parseMemberAccess(p, token, left, true)
 		},
 	)
 }
 
-func parseMemberAccess(p *parser, left ast.Expression, optional bool) ast.Expression {
+func parseMemberAccess(p *parser, token lexer.Token, left ast.Expression, optional bool) ast.Expression {
 
-	p.skipSpaceAndComments(true)
+	// Whitespace after the '.' (dot token) is not allowed.
+	// We parse it anyways and report an error
+
+	if p.current.Is(lexer.TokenSpace) {
+		errorPos := p.current.StartPos
+		p.skipSpaceAndComments(true)
```

Comment:
We use `parseTrivia` now for this no?

Reply:
`skipSpaceAndComments` is just a helper that calls `parseTrivia`, which @zhangchiqing suggested, so all the code can stay as-is and it's shorter than `parseTrivia(parseTriviaOptions{...`